src/OpenTracing/TracerSpanBuilderExtensions.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace OpenTracing
+{
+    using System.Linq;
+
+    public static class TracerSpanBuilderExtensions
+    {
+        public static ISpanBuilder BuildSpan(ITracer tracer, string operationName)
+        {
+            return new SpanBuilder(tracer, operationName);
+        }
+
+        private sealed class SpanBuilder : ISpanBuilder
+        {
+            private readonly ITracer tracer;
+
+            private string operationName;
+            private ICollection<Tuple<string, ISpanContext>> references;
+            private DateTimeOffset? explicitStartTime;
+            private IDictionary<string, object> tags;
+
+            public SpanBuilder(ITracer tracer, string operationName)
+            {
+                this.tracer = tracer;
+                this.operationName = operationName;
+                this.references = new List<Tuple<string, ISpanContext>>();
```

Comment:
Initial sizes for the list and dictionary should be supported for optimization purposes.